### PR TITLE
Introduce a @nivo/controls package

### PR DESCRIPTION
**Pull request by plouc**
The purpose of this PR is to provide similar controls (UI & state management) to those you have in the interactive demo in the form of a package, something similar to this (not a final version):

nivo_controls

It also adds an interactive documentation about how to use the package to the website in a new "Controls" guide, which will looks like this (also not necessarily a final version):

Screen Shot 2022-01-28 at 6 41 58

I would also like to provide an app to easily create nivo charts at some point, and having this package would definitely help building it. Could be something similar to this (early prototype, not sure it'll ever be released):

Screen Shot 2022-01-28 at 6 55 45

The PR also contains some unrelated changes:
- Format the codesandbox examples with Prettier.
- Update the codesandbox examples `@nivo` packages versions to the latest version.
- Adds a new `@nivo/exporter` package to export SVG charts to `.svg` or images.
- Update `react-select` to the latest version (used for the website), shouldn't impact packages.
- Migrate the `Colors` guide available in the website to TypeScript.

### CODE REVIEW

**Comment by tebeco** [comment]
TLDR: will remove this comment if needed if a description is added to this PR or an issue so that it don't polute the discussion of that PR

Is it too soon to put a short description at the goal it aim to resolve @plouc ?
Or link the associated issue(s) that linked to that PR ?

image

Goal
* do not need to open file diff / commit to understand what's under the PR
* do not need to manually open "file diff" when "hey remember we changed that with #1890 
  that last point literally happened (not sure if Nivo or another npm lib)
  => PR with an empty description
  => someone replied "you had to look better at the code diff"
  => was a breaking change

**Comment by plouc** [comment]
@tebeco I cannot disagree about this, that's something I tried to do for recent PRs such as https://github.com/plouc/nivo/pull/1880.

Being a side project and the main (and only at the moment) maintainer, I tend to work in spikes when I can find some time, this also means that PRs can become pretty huge pretty quickly, I often add some unrelated changes to the codebase when I think it could benefit the project.

I've added a description to the PR, explaining the goal and listing the unrelated changes. There are no issues related to this PR, it's a new feature which I think could benefit users, providing the same kind of controls you can find in the interactive demo in the form of a package.

Some breaking changes are and will be introduced in the lib without new major versions being used until the lib reaches a V1,  there are a few major tasks to complete before reaching it:

- Complete the TypeScript migration
- Replace `react-motion` with `react-spring`
- Remove `recompose` from every packages
- Homogenize packages' architecture

That being said, it is indeed important to try identify and document them 😅

**Comment by tebeco** [comment]
thx A LOT for all that comitment @plouc (that being Nivo or updating that PR description ;) )

> Complete the TypeScript migration

that is definitly great news
